### ISSUE TITLE
Import starting with double underscore is renamed by "organize imports" command

### ISSUE THREAD

**Issue by cspotcode** [buehler/typescript-hero#320]
#### System information

- OS: Windows 10
- TypeScript Hero version: 1.5.0
- VSCode version: 1.17.1

#### Description

The "organize imports" command erroneously changes the local name of imports within the import statement when the local identifier starts with two underscores.  It adds a third underscore; it should not change the name at all.

##### Steps to reproduce

1. Create an empty directory.
1. Add the `index.ts` below:
1. Run `npm install @types/node typescript`.
1. Open VSCode in that directory.
1. Open index.ts
1. Invoke the "organize imports" command.

The imports on line 1 and 2 have been renamed to have an extra underscore.

```
// index.ts
import { join as __join } from 'path';
import * as __path from 'path';
__path.join('what', 'ever');
__join('what', 'ever');
```

**Comment by shobhitg** [comment]
This issue is happening due to typescript's own code:

https://github.com/Microsoft/TypeScript/blob/d7be61a569608dba24d2b384e6fee12533a36736/src/compiler/utilities.ts#L383

```
    // Add an extra underscore to identifiers that start with two underscores to avoid issues with magic names like '__proto__'
    export function escapeLeadingUnderscores(identifier: string): __String {
        return (identifier.length >= 2 && identifier.charCodeAt(0) === CharacterCodes._ && identifier.charCodeAt(1) === CharacterCodes._ ? "_" + identifier : identifier) as __String;
    }
```

I don't have a solution, but just noting down my findings.

**Comment by cspotcode** [comment]
It looks like the compiler consistently adds an extra underscore to every
single identifier that starts with at least 2 underscores.  And it has an
`unescapeLeadingUnderscores` function to do the reverse.  I assume we need
to figure out where TypeScript hero, or more likely TypeScript-parser, is
forgetting to unescape.

**Comment by buehler** [comment]
Fixed by #327
